includes/class-sensei.php

```diff
@@ -918,80 +919,166 @@ public function load_class( $class_name = '' ) {
 
-		$statuses = array( '' ); // Default to the WP normal comments
-		// WC excludes these so exclude them too
-		$wc_statuses_to_exclude = array( 'order_note', 'webhook_delivery' );
-		$stati                  = $wpdb->get_results( "SELECT comment_type FROM {$wpdb->comments} GROUP BY comment_type", ARRAY_A );
-		foreach ( (array) $stati as $status ) {
-			if ( 'sensei_' != substr( $status['comment_type'], 0, 7 ) &&
-				! in_array( $status['comment_type'], $wc_statuses_to_exclude ) ) {
-				$statuses[] = $status['comment_type'];
+		// Subtract Sensei's comment counts from all the comment counts.
+		foreach ( array_keys( (array) $comment_counts ) as $count_type ) {
+			if ( isset( $sensei_counts[ $count_type ] ) ) {
+				$comment_counts->{$count_type} = max( 0, (int) $comment_counts->{$count_type} - $sensei_counts[ $count_type ] );
 			}
 		}
-		$where = "WHERE comment_type IN ('" . join( "', '", array_unique( $statuses ) ) . "')";
 
-		if ( $post_id > 0 ) {
-			$where .= $wpdb->prepare( ' AND comment_post_ID = %d', $post_id );
+		return $comment_counts;
+	}
+
+	/**
+	 * Clear the comment count cache.
+	 *
+	 * @param int $post_id Post ID.
+	 */
+	public function flush_comment_counts_cache( $post_id ) {
+		$post_id = (int) $post_id;
+
+		$post_transient_id = self::COMMENT_COUNT_TRANSIENT_PREFIX . $post_id;
+		delete_transient( $post_transient_id );
+
+		$all_transient_id = self::COMMENT_COUNT_TRANSIENT_PREFIX . '0';
+		delete_transient( $all_transient_id );
+	}
+
+	/**
+	 * Get the Sensei related comment counts by comment_approved but use caching.
+	 *
+	 * @param int $post_id Post ID.
+	 *
+	 * @return array
+	 */
+	private function get_sensei_comment_counts( $post_id ) {
+		$post_id = (int) $post_id;
+
+		$transient_id = self::COMMENT_COUNT_TRANSIENT_PREFIX . $post_id;
+		$stats        = get_transient( $transient_id );
+
+		if ( ! $stats || ! is_array( $stats ) ) {
+			$stats = $this->get_sensei_comment_counts_direct( $post_id );
+
+			set_transient( $transient_id, $stats );
+		}
+
+		return $stats;
+	}
+
+	/**
+	 * Get the Sensei related comment counts by comment_approved.
+	 *
+	 * @param  int $post_id Post ID.
+	 *
+	 * @return array
+	 */
+	private function get_sensei_comment_counts_direct( $post_id ) {
+		global $wpdb;
+
+		$post_where = '';
+
+		if ( ! empty( $post_id ) ) {
+			$post_where = $wpdb->prepare( 'AND comment_post_ID=%d', $post_id );
 		}
 
-		$count = $wpdb->get_results( "SELECT comment_approved, COUNT( * ) AS num_comments FROM {$wpdb->comments} {$where} GROUP BY comment_approved", ARRAY_A );
 
-		$total    = 0;
-		$approved = array(
-			'0'            => 'moderated',
-			'1'            => 'approved',
-			'spam'         => 'spam',
-			'trash'        => 'trash',
-			'post-trashed' => 'post-trashed',
+		$counts = $wpdb->get_results(
+			"
+					SELECT comment_approved, COUNT(*) AS num_comments
+					FROM {$wpdb->comments}
+					WHERE
+						comment_type IN ( 'sensei_course_status', 'sensei_lesson_status', 'sensei_user_answer' )
```

Comment:
`sensei_user_answer` is just legacy, I think, but thought I should include it. Might be other legacy types we need to include?

Reply:
We could also do something like ` LIKE 'sensei_%'`. I have the impression that this clause would use the index if it exists and would catch all legacy cases.

Reply:
Switched to `LIKE 'sensei_%`